core/shared/src/main/scala/scodec/bits/ByteVector.scala

```diff
@@ -625,6 +625,21 @@ sealed abstract class ByteVector extends BitwiseOperations[ByteVector, Long] wit
   }
 
   /**
+   * Copies as many bytes as possible to the given [[ByteBuffer]], starting from its
+   * current position. This method will not overflow the buffer.
+   *
+   * @param buffer a ByteBuffer to copy bytes to
+   * @return the number of bytes actually copied
+   * @group conversions
+   */
+  final def copyToBuffer(buffer: ByteBuffer): Int = {
+    val limited = take(buffer.remaining.toLong)
```

Comment:
I don't know whether it's required that we do the full `take` here first.
In `copyToArray` we are able to avoid it through some offset calculations.
Can't we do the same here?